src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -0,0 +1,34 @@
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import type { AbstractQueryGenerator } from './query-generator';
+
+export interface QueryInterfaceOptions {
+  sequelize: Sequelize;
+  queryGenerator: AbstractQueryGenerator;
+}
+
+export class AbstractQueryInterfaceTypeScript {
+  protected readonly sequelize: Sequelize;
```

Comment:
These two should remain public for now. It's the only way to access the sequelize instance in migrations (until we rework migrations), and the only way to access queryGenerator altogether (until we expose it on the sequelize instance)